udpTester.py

```diff
@@ -57,206 +71,204 @@ def close(self):
+        )
+
+
+@dataclasses.dataclass
 class udpMetrics:
-    sampleCount = 0
-    maxCount = 0
-    values = []
-    def __init__(self, maxNofSamples):
-        self.reset()
-        self.maxCount = maxNofSamples
-
-    def reset(self):
-        self.sampleCount = 0
-        self.values = []
-
-    def addValue(self, value):
-        assert(self.sampleCount < self.maxCount)
-        if(self.sampleCount < self.maxCount):
-            self.values.append(value)
-            self.sampleCount += 1
-
-    def calculateReport(self, percentile):
-        if( self.sampleCount <= 0 ):
+    max_number_of_samples: int
+    samples: list = dataclasses.field(default_factory=list)
+
+    def append(self, sample):
+        if len(self.samples) < self.max_number_of_samples:
+            self.samples.append(sample)
+
+    def report(self, percentile):
+        # ensure at least one sample
+        if not self.samples:
             return udpMetricsReportItem()
-        else:
-            reportItem = udpMetricsReportItem()
-            range = int((self.sampleCount * percentile)/100)
-            reportItem.percentile  = percentile
-            reportItem.sampleCount = self.sampleCount
-            reportItem.minimum     = self.values[0]
-            if( len(self.values[:range]) <= 0 ):
-                reportItem.average = math.nan
-            else:
-                reportItem.average = statistics.mean(self.values[:range])
-            reportItem.maximum     = self.values[range-1]
-            if( len(self.values[:range]) <= 1 ):
-                reportItem.deviation = math.nan
-            else:
-                reportItem.deviation = statistics.stdev(self.values[:range])
-            return reportItem
 
-    def analyse(self, percentiles):
-        self.values.sort()
-        udpMetricsReport = []
-        for percentile in percentiles:
-            udpMetricsReport.append(self.calculateReport(percentile))
-        return udpMetricsReport
+        count = 1 + math.floor(len(self.samples) * percentile / 100.0 - 0.999)
+        samples = self.samples[:count]
+
+        return udpMetricsReportItem(
+            percentile=percentile,
+            sampleCount=len(samples),
+            totalSampleCount=len(self.samples),
+            minimum=samples[0],
+            average=statistics.mean(samples),
+            maximum=samples[-1],
+            deviation=statistics.stdev(samples) if len(samples) > 1 else math.inf,
+        )
+
+    def reports(self, percentiles):
+        self.samples.sort()
+        return [self.report(percentile) for percentile in percentiles]
+
 
 def transmitter(args):
     print("I am the transmitter")
-    DEFAULT_MSGSIZE    =       100
-    DEFAULT_SLEEPTIME  =        20
-    DEFAULT_TOTNOFMSGS =      1000
-    DEFAULT_PORTNR     =     10350
-    DEFAULT_PACKETSIZE =      1300
-    DEFAULT_LOSSINESS  =         0
-    DEFAULT_MULTI_TTL  =        64
+    DEFAULT_MSGSIZE = 100
```

Comment:
I wrote it like this to line up the numbers for readability. Do you think it is not a good idea?